pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -280,4 +288,15 @@ private QueryPlan toDispatchablePlan(RelRoot relRoot, PlannerContext plannerCont
   private HintStrategyTable getHintStrategyTable() {
     return PinotHintStrategyTable.PINOT_HINT_STRATEGY_TABLE;
   }
+
+
+  private Set<String> getTableNamesFromRelRoot(RelNode relRoot) {
+    Set<String> tableNames = new HashSet<>();
+    List<String> qualifiedTableNames = RelOptUtil.findAllTableQualifiedNames(relRoot);
+    for (String qualifiedTableName : qualifiedTableNames) {
+      String tableName = qualifiedTableName.replaceAll("^\\[(.*)\\]$", "$1");
```

Comment:
Can you elaborate why this regex is needed ?

Reply:
Good callout. I added a comment in the code explaining why it is needed.